File: sorting/problem324/problem324.py
from typing import List


class Solution:
    def wiggleSort(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # Sort, then place smaller half at even indices and larger half at odd indices,
        # both halves taken from the end inward so duplicates do not sit next to each other.
        nums.sort()
        n = len(nums)
        mid = (n + 1) // 2
        left = nums[:mid][::-1]
        right = nums[mid:][::-1]
        nums[::2] = left
        nums[1::2] = right

File: sorting/problem324/problem324.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    void wiggleSort(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = static_cast<int>(nums.size());
        int mid = (n + 1) / 2;
        vector<int> left(nums.begin(), nums.begin() + mid);
        vector<int> right(nums.begin() + mid, nums.end());
        reverse(left.begin(), left.end());
        reverse(right.begin(), right.end());
        for (int i = 0; i < static_cast<int>(left.size()); ++i) {
            nums[2 * i] = left[i];
        }
        for (int i = 0; i < static_cast<int>(right.size()); ++i) {
            nums[2 * i + 1] = right[i];
        }
    }
};
